lib/vmdb/config.rb

```diff
@@ -7,19 +7,23 @@ class Config
 
     require_relative 'configuration_encoder'
 
-    def self.clone_auth_for_log(auth)
-      log_auth = auth.deep_clone
-      [:bind_pwd, :amazon_secret].each do |key|
-        log_auth[key] = '********' if log_auth.key?(key)
-        if log_auth.key?(:user_proxies)
-          log_auth[:user_proxies].each do |p|
-            p[key] = '********' if p.key?(key)
-          end
-        end
+    def self.deep_clean_auth_for_log(log_auth)
+      log_auth.each do |key, val|
+        log_auth[key] = deep_clean_auth_for_log(val) if val.kind_of?(Hash)
+
+        log_auth[key] = "********" if Vmdb::ConfigurationEncoder::PASSWORD_FIELDS.include?(key.to_s)
+
+        log_auth[key].each do |p|
+          p.keys.each { |key2| p[key2] = "********" if Vmdb::ConfigurationEncoder::PASSWORD_FIELDS.include?(key2.to_s) }
+        end if [:user_proxies].include? key
       end
       log_auth
```

Comment:
Late night refactoring for fun here :smile: 
What do you think of this version?

```ruby
def self.deep_clean_auth_for_log(log_auth)
  return log_auth unless log_auth.kind_of? Hash
  log_auth.each do |key, value|
    log_auth[key] = case value
                      when Hash
                        deep_clean_auth_for_log(value)
                      when Array
                        value.map{|key| deep_clean_auth_for_log(key)}
                      else
                        key.to_s.in?(Vmdb::ConfigurationEncoder::PASSWORD_FIELDS) ? "********" : value
                    end
  end
end
```